doc/adr/0009-time-graph-search-functionality.md

```diff
@@ -0,0 +1,166 @@
+
+Here's an example Sequence Diagram for the search request above:
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant timegraph as timegraph-output-component
+    participant server as Trace Server
+    User->>timegraph: enter search filter regex ("openat")
+    timegraph->>server: POST /experiments/{expUUID}/outputs/timegraph/{outputId}/states
+    Note over timegraph,server: requested_timerange=[current_timerange]
+    Note over timegraph,server: requested_items=[selected_entry_Ids]
+    Note over timegraph,server: regex_map_filters={"1": ["openat"]}
+    Note over timegraph,server: full_search=true
+    server->>timegraph: 200: TimelineChart.TimeGraphModel
+    timegraph->>timegraph: TimeGraphChartProviders.stateStyleProvider updates state style
+    timegraph->>timegraph: Update timegraph
+```
+
+I've attached a link to a prototype in trace-extension that makes use of the aforementioned parameters to  search time events in time graph views:
+- [https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886][pull-886]
+- This prototype adds a search-box to the timegraph-output-component to search for events. The filter is applied and unsuccessful time events are dimmed.
+- At the moment its only looking at sampled search, we will add full-search functionality in the near future.
+- The search-box only takes in a single search string - we will add the ability to apply multiple filters through future patches.
+- This might not be the final UI, but it gives the idea on how to achieve it.
+ 
+Even through the current TSP can support search functionality, the use of `regex_map_filters` does not seem to be very intuitive, as a client developer would not be sure of what to pass and what the expected response should be. A suggestion is to rename `regex_map_filters` to `filter_query_parameters`. `filter_query_parameters` takes a `filter_expressions_map`, which contains the search expressions as shown in `regex_map_filters`. 
+
+There is also some confusion with the `full_search` flag, and a suggestion is to change it to `strategy`. `strategy` would accept 2 possible values `sampled` and `deep`. The `strategy` flag is an optional parameter within `filter_query_parameters`, and if `strategy` is omitted then `sampled` search would be the default value.
+The following example does the same query given above but uses `filter_query_parameters` and `strategy` instead:
+```
+{
+  "parameters": {
+    "requested_timerange": {
+      "start": 111111111,
+      "end": 222222222,
+      "nbTimes": 1920
+    },
+    "requested_items": [
+      1,
+      2
+    ],
+    "filter_query_parameters": {
```

Comment:
Does anyone think that this proposal should also consider / compare to the possibility of using graphQL to make such "filtering queries"? I am not sure if [adr 002] was meant to look into this or not, but I think that at least a few words regarding if graphQL would be adapt to do similar "filter queries" or not could be added

[adr 002]: https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/master/doc/adr/0002-GraphQL.md

Reply:
No, I don't think GraphQL should be considered for "filtering queries". ADR 002 investigated GraphQL as replacement of the current TSP implementation.